python/ql/lib/semmle/python/frameworks/data/internal/ApiGraphModelsSpecific.qll

```diff
@@ -92,12 +92,19 @@ API::Node getExtraSuccessorFromInvoke(API::CallNode node, AccessPathToken token)
   token.getName() = "Instance" and
   result = node.getReturn()
   or
-  token.getName() = "Argument" and
-  token.getAnArgument() = "self" and
-  result = node.getSelfParameter()
-  or
-  token.getName() = "Argument" and
-  exists(string arg | arg + ":" = token.getAnArgument() | result = node.getKeywordParameter(arg))
+  token.getName() = ["Argument", "Parameter"] and
```

Comment:
Yep, that was an oversight from me.  
I've added the tests.  

Currently they are aliases of each other, and they both include the self parameter.  
I'm not sure if we should change that?

Reply:
Yes that is a bit strange. I think we might want to discuss with Asger and Ruby team on what the right approach here is.

I think it's strange that for a function where all arguments can be given as either positional or keyword arguments, `Parameter[any-named]` and `Parameter[0..]` should give different results.

I'm not sure whether `Parameter[any]` should include `Parameter[self]` or not though :thinking: My initial reaction is that it probably shouldn't be included, since my guess is that you would be interested in any parameter (positional-only/normal/keyword-only) that receives values passed as arguments to the function -- but I'm not 100% sure on this.

Reply:
> I'm not sure whether Parameter[any] should include Parameter[self] or not though thinking My initial reaction is that it probably shouldn't be included, since my guess is that you would be interested in any parameter (positional-only/normal/keyword-only) that receives values passed as arguments to the function -- but I'm not 100% sure on this.  

I've (for now) gone for the approach of excluding the self parameter from `Parameter[any]` and `Parameter[any-named]` by excluding the `self` parameter from the `keywordArgument` edge in API-graphs.   

This also makes it consistent with the argument side, which never included the self parameter 
(also know as the receiver or base of a method call).